[DOC] Tokenizers - Character group (#8350)
opensearch-project/documentation-website: _analyzers/tokenizers/character-group-tokenizer.md

---
layout: default

> title: Character Group Tokenizer

Reviewer: ```suggestion
title: Character group tokenizer
```

parent: Tokenizers
nav_order: 20
has_children: false
has_toc: false
---

# Character group tokenizer

> The character group tokenizer is designed to segment text into tokens based on the presence of specific characters. This tokenizer is ideal for scenarios where a straightforward tokenization approach is required, avoiding the complexity and overhead associated with pattern-based tokenizers.

Reviewer: ```suggestion
The character group tokenizer is a simple text segmentation tool that splits text into tokens based on the presence of specific characters. This tokenizer is ideal for scenarios where a simple tokenization method is required, avoiding the complexity and overhead associated with pattern-based tokenizers.
```

> The character group tokenizer accepts the following parameters:

Reviewer: ```suggestion
The character group tokenizer accepts the following parameters:

```

---
layout: default
title: Character group tokenizer
parent: Tokenizers
nav_order: 20
has_children: false
has_toc: false
---

# Character group tokenizer

The character group tokenizer is designed to segment text into tokens based on the presence of specific characters. This tokenizer is ideal for scenarios where a straightforward tokenization approach is required, avoiding the complexity and overhead associated with pattern-based tokenizers.

The character group tokenizer accepts the following parameters:

> 1. `tokenize_on_chars`: Specifies a set of characters on which the text should be tokenized. When any character from this set is encountered, a new token is created. For example, single characters `(e.g., -, @)` and character classes such as `whitespace`, `letter`, `digit`, `punctuation`, and `symbol`.

Reviewer: ```suggestion
1. `tokenize_on_chars`: Specifies a set of characters on which the text should be tokenized. The tokenizer creates a new token upon encountering any character from the specified set. For example, single characters `(e.g., -, @)` and character classes such as `whitespace`, `letter`, `digit`, `punctuation`, and `symbol`.
```

> 1. `tokenize_on_chars`: Specifies a set of characters on which the text should be tokenized. The tokenizer creates a new token upon encountering any character from the specified set, for example, single characters `(e.g., -, @)` and character classes such as `whitespace`, `letter`, `digit`, `punctuation`, and `symbol`.

Reviewer 2: also would be good to add that this tokenizer accepts escape characters

> 4. `max_token_length`: This parameter defines the maximum length allowed for a token. If a token exceeds this specified length, it will be split at intervals defined by `max_token_length`. The default value is `255`.

Reviewer: ```suggestion
2. `max_token_length`: This parameter defines the maximum length allowed for a token. If a token exceeds this specified length, it will be split at intervals defined by `max_token_length`. The default value is `255`.
```

> 2. `max_token_length`: This parameter defines the maximum length allowed for a token. If a token exceeds this specified length, it will be split at intervals defined by `max_token_length`. The default value is `255`.

Reviewer: ```suggestion
2. `max_token_length`: Defines the token's maximum length. If a token exceeds this specified length, it is split at intervals defined by the parameter. Default is `255`.
```

> ## Example of the character group tokenizer

Reviewer: ```suggestion
## Example: Using the character group tokenizer
```

> We can tokenize the on characters such as `whitespace`, `-` and `:`.

Reviewer: ```suggestion
To tokenize the on characters such as `whitespace`, `-` and `:`, see the following example request:
```

```json
POST _analyze
{
  "tokenizer": {
    "type": "char_group",
    "tokenize_on_chars": [
      "whitespace",
      "-",
      ":"
    ]
  },
  "text": "Fast-cars: drive fast!"
}
```
{% include copy-curl.html %}

> By analyzing the text "Fast-cars: drive fast!", we can see the specified characters have been removed: 

Reviewer: ```suggestion
The following response shows that the specified characters have been removed: 
```

```

> Fast cars drive fast

Reviewer 2: This returns 

```
Fast cars drive fast!
```
